ambry-cloud/src/main/java/com/github/ambry/cloud/CloudStorageCompactor.java

```diff
@@ -212,6 +214,28 @@ public int compactPartition(String partitionPath, String fieldName, long querySt
 
     // Iterate until returned list size < limit, time runs out or we get shut down
     int totalPurged = 0;
+    long chunkTimeRange = TimeUnit.DAYS.toMillis(queryBucketDays);
+    if (queryEndTime - queryStartTime > chunkTimeRange) {
+      logger.debug("Dividing compaction query for {} into buckets of {} days", partitionPath, queryBucketDays);
+      long chunkedStartTime = queryStartTime;
+      while (chunkedStartTime < queryEndTime) {
+        long chunkedEndTime = Math.min(chunkedStartTime + chunkTimeRange, queryEndTime);
+        int numPurged = compactPartition(partitionPath, fieldName, chunkedStartTime, chunkedEndTime, timeToQuit);
```

Comment:
We should move the logic that actually does the compaction (which is basically everything outside the if statement in line 218), in a separate method of its own, unless we have reason not to do so. Makes code easy to read, keeps methods small and separates out the control part of compaction logic from the actual compaction.

Reply:
I had it that way initially but then consolidated it into single method.  I can change it back if it helps readability.